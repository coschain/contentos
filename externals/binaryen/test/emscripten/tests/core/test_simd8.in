#include <stdio.h>
#include <xmmintrin.h>

char buffer[21];

int main() {
  _mm_storeu_ps((float*)&buffer[1], (__m128){0.1, 2.3, 4.5, 6.7});
  __m128 y = _mm_loadu_ps((float*)&buffer[5]);

  float __attribute__((__aligned__(16))) ar[4];
  *(__m128 *)&ar = y;
  for (int i = 0; i < 4; i++) {
    printf("%f\n", ar[i]);
  }

  return 0;
}
