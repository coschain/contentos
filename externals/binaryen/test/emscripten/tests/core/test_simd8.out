2.300000
4.500000
6.700000
0.000000
